src/lcgs/ir/intermediate.rs

```diff
@@ -76,11 +77,11 @@ impl IntermediateLCGS {
             let symb = self.symbols.get(symb_id).unwrap();
             if let DeclKind::StateVar(var) = &symb.kind {
                 let value = {
-                    let size = var.ir_range.end() - var.ir_range.start() + 1;
+                    let size = (var.ir_range.end() - var.ir_range.start() + 1) as usize;
```

Comment:
I might ask stupid but i am not sure why this is `as usize` and not `as i32` maybe one can elaborate?

Reply:
`end()` will always be greater than `start()`, so we are guaranteed to get a positive number, but I then also cast it to `usize` since I a few lines later do an operation, where both operands must be of the same type. The other operand is of type `usize` and since `usize` > `i32` we cast upwards to `usize` to not lose precision.